### ISSUE TITLE
Media's url(…) should offer include_host-Parameter

### ISSUE THREAD

**Issue by MaZderMind** [getgrav/grav#894]
A [Media's `url(…) Method](https://learn.getgrav.org/content/media#url) does not offer a `include_host`-Parameter like [the Page-Object does](https://learn.getgrav.org/themes/theme-vars#urlinclude_host-false).

~~Furthermore there seems no other way to add the Hostname to a Media's URL.~~

**Comment by MaZderMind** [comment]
Skip the last Paragraph, one can prepend `uri.base`. But the API should closer mimic the Pages' API which offers this option.

**Comment by rhukster** [comment]
Pretty sure this already takes into account the system.yaml based `absolute_urls` setting.

**Comment by MaZderMind** [comment]
Yes, probably. I needed it here for a Pinterest sharing link, which needs the hostname but just in this place.

**Comment by flaviocopes** [comment]
Can you explain the use case in details, to understand what's the problem and how did you expect it to work?

**Comment by rhukster** [comment]
I think he programmatically want's to flag a particular media URL to use full URL rather than using whatever the default style is.

so:

```
{{ page.media['someimage.jpg'].url(true) }}
```

To provide a full `http://yourhost.com/yourpage/someimage.jpg` type URL for this media item.

**Comment by MaZderMind** [comment]
@rhukster exactly. My Use-Case currently looks like this:

```twig
{% set url = uri.url(true) | url_encode %}
{% set desc = page.title | url_encode %}
{% set image = (uri.base ~ (page.media.images | first).url) | url_encode %}

<a class="pinterest" href="https://pinterest.com/pin/create/button/?url={{ url }}&amp;media={{ image }}&amp;description={{ desc }}" data-width="1200" data-height="630">
	Pinterest
</a>
````

I had expected for `media..url` to have a `include_host` parameter, symmetrical to `uri.url`.

**Comment by flaviocopes** [comment]
Ok got it, you can do via `absolute_urls: true`, but turns on for all images.

**Comment by rhukster** [comment]
Added for Grav 2.1.0. A media file's `url()` takes a second argument that prepends the scheme and host, mirroring `page.url(true)`. So `page.media['photo.jpg'].url(true, true)` gives a full URL for one image where a Pinterest link, Open Graph tag or feed needs it, without turning on `absolute_urls` for the whole site. Works for resized derivatives too.
